### PR TITLE
Added API throttling to RDS module

### PR DESCRIPTION
**Pull request by cooniur**
##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
AWS RDS module

##### ANSIBLE VERSION
```
$ ansible --version
ansible 2.2.0.0
  config file =
  configured module search path = Default w/o overrides
```

##### SUMMARY
```
Added a new parameter `api_throttling` to RDS module to allow specifying a time before calling AWS API to avoid throttling issue. 

Fixes my bug report in Issue #143 (see my comment there).
```

### CODE REVIEW

**Comment by cooniur** [comment]
How to fix the following error?
```
2016-12-03 07:04:05 ERROR: version_added for new option (aws_throttling) should be 2.3. Currently 0.0
```

File: cloud/amazon/rds.py [hunk 1]
```diff
@@ -647,8 +652,9 @@ def get_data(self):
 
 def await_resource(conn, resource, status, module):
     wait_timeout = module.params.get('wait_timeout') + time.time()
+    api_throttling = module.params.get('api_throttling')
```
Comment (ryansb) [inline cloud/amazon/rds.py:655]: Instead of manually doing retries, use [AWSRetry](https://github.com/ansible/ansible-modules-core/blob/db66833a80bcd7f0d33ccbfd6555d69c5ae88003/cloud/amazon/cloudformation.py#L317) which is now part of the EC2 module_utils. It got added early in 2.3, and should cut down on duplicated retry/backoff logic.

File: cloud/amazon/rds.py [hunk 2]
```diff
@@ -292,9 +297,9 @@
      wait: yes
      wait_timeout: 600
      tags:
-         Name: pg1_test_name_tag 
+         Name: pg1_test_name_tag
   register: rds
-
```
Comment (ryansb) [inline cloud/amazon/rds.py:297]: Generally we try to avoid making whitespace changes to areas you aren't editing. If possible, can you set your editor to not automatically do this?

File: .gitignore [hunk 3]
```diff
@@ -51,3 +51,5 @@ venv
 Vagrantfile
 .vagrant
 ansible.egg-info/
+# Visual Studio Code
+.vscode
```
Comment (ryansb) [inline .gitignore:55]: Please do this in a separate PR, since it's not related to the RDS module.

**Review by ryansb — CHANGES_REQUESTED** [review]
Thanks for this PR. We have a decorator called AWSRetry to help with this sort of logic, can you use it instead?

File: cloud/amazon/rds.py (hunk 2 above)
Comment (cooniur) [reply cloud/amazon/rds.py:297]: OK,  will revert the whitespace changes.

File: cloud/amazon/rds.py (hunk 1 above)
Comment (cooniur) [reply cloud/amazon/rds.py:655]: @ryansb I'd love to dig more about the AWSRetry. But for the simplicity & quickness of fixing the api throttling issue, can we accept this for now? (and of course, do the AWSRetry conversion in a separate PR since the RDS module hasn't started using it yet anyway).

File: .gitignore (hunk 3 above)
Comment (cooniur) [reply .gitignore:55]: OK, will do it in a separate PR.

**Comment by cooniur** [comment]
Hi @ryansb , hopefully my last comments make sense to you. Also I just wanted to let you know that I still remember this PR and will just need to find some time to make the requested changes.

Thanks!
